File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/types/message/ReadVariableValueRequest.java
package org.opcfoundation.webserver.types.message;

import org.opcfoundation.webserver.types.UaObjectId;
import org.opcfoundation.webserver.types.UaChildId;

import java.util.Set;

public class ReadVariableValueRequest {
    private final UaObjectId objectId;
    private final Set<UaChildId> variableIds;

    public ReadVariableValueRequest(
            UaObjectId objectId,
            Set<UaChildId> variableIds)
    {
        this.objectId = objectId;
        this.variableIds = variableIds;
    }

    public UaObjectId getObjectId() {
        return objectId;
    }

    public Set<UaChildId> getVariableIds() {
        return variableIds;
    }
}

File: opcua-webservice-node/src/server/service/message/reactiveobject/ReadVariableValueRequest.ts
import { UaChildId, UaObjectId } from "../../../types";

export class ReadVariableValueRequest {
    private readonly _objectId: UaObjectId;
    private readonly _variableIds: Set<string>;

    constructor(
        objectId: UaObjectId,
        variableIds: Set<string>,
    ) {
        this._objectId = objectId;
        this._variableIds = variableIds;
    }

    get objectId(): UaObjectId {
        return this._objectId;
    }

    get variableIds(): Set<string> {
        return this._variableIds;
    }
}